frontend/app/mappers/personal-information-service-mappers.server.ts

```diff
@@ -0,0 +1,228 @@
+    clientId: personalInformationApiResponse.BenefitApplication.Applicant?.ClientIdentification?.filter((clientInfoDto) => clientInfoDto.IdentificationCategoryText === 'Client ID').at(0)?.IdentificationID,
+    clientNumber: personalInformationApiResponse.BenefitApplication.Applicant?.ClientIdentification?.filter((clientInfoDto) => clientInfoDto.IdentificationCategoryText === 'Client Number').at(0)?.IdentificationID,
+    birthDate: personalInformationApiResponse.BenefitApplication.Applicant?.PersonBirthDate.dateTime,
+    firstName: personalInformationApiResponse.BenefitApplication.Applicant?.PersonName?.at(0)?.PersonGivenName?.at(0),
+    lastName: personalInformationApiResponse.BenefitApplication.Applicant?.PersonName?.at(0)?.PersonSurName,
+    emailAddress: personalInformationApiResponse.BenefitApplication.Applicant?.PersonContactInformation.at(0)?.EmailAddress.at(0)?.EmailAddressID,
+    maritalStatusId: personalInformationApiResponse.BenefitApplication.Applicant?.PersonMaritalStatus.StatusCode.ReferenceDataID,
+    homeAddress: homeAddressList
+      ?.map((aHomeAddress) => ({
+        streetName: aHomeAddress.AddressStreet.StreetName,
+        secondAddressLine: aHomeAddress.AddressSecondaryUnitText,
+        cityName: aHomeAddress.AddressCityName,
+        provinceTerritoryStateId: aHomeAddress.AddressProvince.ProvinceCode.ReferenceDataID,
+        countryId: aHomeAddress.AddressCountry.CountryCode.ReferenceDataID,
+        postalCode: aHomeAddress.AddressPostalCode,
+      }))
+      .at(0),
+    mailingAddress: mailingAddressList
+      ?.map((aMailingAddress) => ({
+        streetName: aMailingAddress.AddressStreet.StreetName,
+        secondAddressLine: aMailingAddress.AddressSecondaryUnitText,
+        cityName: aMailingAddress.AddressCityName,
+        provinceTerritoryStateId: aMailingAddress.AddressProvince.ProvinceCode.ReferenceDataID,
+        countryId: aMailingAddress.AddressCountry.CountryCode.ReferenceDataID,
+        postalCode: aMailingAddress.AddressPostalCode,
+      }))
+      .at(0),
+
+    primaryTelephoneNumber: personalInformationApiResponse.BenefitApplication.Applicant?.PersonContactInformation.at(0)?.TelephoneNumber.find((phoneNumber) => phoneNumber.TelephoneNumberCategoryCode.ReferenceDataName === 'Primary')?.FullTelephoneNumber
+      .TelephoneNumberFullID,
+    alternateTelephoneNumber: personalInformationApiResponse.BenefitApplication.Applicant?.PersonContactInformation.at(0)?.TelephoneNumber.find((phoneNumber) => phoneNumber.TelephoneNumberCategoryCode.ReferenceDataName === 'Alternate')?.FullTelephoneNumber
+      .TelephoneNumberFullID,
+    preferredLanguageId: personalInformationApiResponse.BenefitApplication.Applicant?.PreferredMethodCommunicationCode?.ReferenceDataID,
+  };
+}
+
+export function toPersonalInformationDB(personalInformation: PersonalInfo): ToPersonalInformationDBArgs {
```

Comment:
This function and anything related to mapping to the mock database structure should be moved to the Power Platform API mock.

Reply:
moved